example-deployment/README.md

```diff
@@ -0,0 +1,329 @@
+
+```bash
+.bin/minikube kubectl -- apply -f k8s/keycloak.yaml
+```
+
+This requires internet connection, see [Deploying in an airgapped network](#deploying-in-an-airgapped-network) if you are in an airgap network. tldr:
+
+```bash
+# connected to internet
+docker pull quay.io/keycloak/keycloak:12.0.4
+
+# connected to airgap network
+.bin/minikube load quay.io/keycloak/keycloak:12.0.4
+.bin/minikube kubectl -- apply -f k8s/keycloak.yaml
+```
+
+Use kubectl to wait for keycloak to be ready
+
+```bash
+.bin/minikube kubectl -- get deployment
+```
+
+When keycloak is ready, test it out by going to https://example.com/auth. The superuser created by this deployment is `user=admin,password=admin`.
+
+### Configuration
+
+For this example, we will
+
+1. Create a `rmf-web` realm.
+1. Create a `dashboard` client.
+1. Add https://example.com to the list of allowed origins.
+1. Create an example user with user=example password=example.
+
+For brevity, you can use the provided script to automate the process.
+
+```bash
+node keycloak-tools/bootstrap-keycloak.js
+```
+
+You may check out the keycloak docs if you would like to learn more about it's features.
+
+### Get the keycloak cert
+
+Keycloak implements the openid-connect standard, by default the JWT is signed with an auto generated RSA key. We will need the cert so that `rmf-web` backend services can verify the token. You can use the admin console to get it or use the provided script
+
+```bash
+node keycloak-tools/get-cert.js > keycloak.pem
+```
+
+If you wish to, you may check the cert with
+
+```bash
+openssl x509 -in keycloak.pem -noout -text
+```
+
+get the public key from the cert
+
+```bash
+openssl x509 -in keycloak.pem -pubkey -noout -out jwt-pub-key.pub
+```
+
+**NOTE: deleting the persistent volume used by postgres will delete the database, the next time you start keycloak it will start from a fresh state, meaning the realm we created will be gone and the public key will be different.**
+
+### Add the cert to kubernetes
+
+In order to use the cert, we will add it as a configmap to kubernetes
+
+```bash
+.bin/minikube kubectl -- create configmap jwt-pub-key --from-file=jwt-pub-key.pub
+```
+
+note: to update the configmap, delete it first and re-create it
+
+```bash
+.bin/minikube kubectl -- delete configmap jwt-pub-key
+.bin/minikube kubectl -- create configmap jwt-pub-key --from-file=jwt-pub-key.pub
+```
+
+## ros2-bridge
+
+### Build minimal RMF image
+
+We will need a minimal base image containing all the RMF messages. This image will be used to build the ros2-bridge image.
+
+How you get this image will vary depending what version of RMF, and what extensions you are deploying it with. It is important that ros2-bridge is using the exact same message definitions as used by the deployment of RMF. For this example, we assume that you built rmf_demos from source using the main branch, so we will build the messages from source as well.
+
+Get RMF source code
```

Comment:
You can't mount volumes while building images. Also, you should not be "syncing" builds in a deployment, it should be using a pinned version so that every deployment is the same.